server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -240,6 +261,9 @@ void deleteClusterMetadata(
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleGlobalMetadataPaths));
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleIndexMetadataPaths));
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleManifestPaths));
+            if (remoteRoutingTableService.isPresent()) {
+                remoteRoutingTableService.get().deleteStaleIndexRoutingPaths(new ArrayList<>(staleIndexRoutingPaths));
```

Comment:
Why do we need this conversion from Set to List?

Reply:
The function `deleteBlobsIgnoringIfNotExists` accepts a `List<String>` as a parameter and is called within the `deleteStaleIndexRoutingPaths` function.